presto-main/src/main/java/io/prestosql/sql/planner/PlanOptimizers.java

```diff
@@ -407,7 +405,7 @@ public PlanOptimizers(
                 new CheckSubqueryNodesAreRewritten(),
                 new StatsRecordingPlanOptimizer(
                         optimizerStats,
-                        new PredicatePushDown(metadata, typeAnalyzer, false)),
```

Comment:
Yup. I think in this case it might be relatively easy to remove this dependency, so that dynamic filters don't change existing plans. I just didn't want to dig into this particular issue